geonode/maps/api/views.py

```diff
@@ -156,9 +156,13 @@ def perform_update(self, serializer):
         map_layers = serializer.validated_data.get("maplayers", [])
         tabular_collection = all(("tabular" in layer.dataset.subtype) for layer in map_layers)
 
-        instance = serializer.save(
-            subtype="tabular-collection" if tabular_collection else None,
-        )
+        # incomming maplayer tabular check
+        map_layers = serializer.validated_data.get("maplayers", [])
+        if len(map_layers) > 0:
```

Comment:
Good point: An empty `all` returns `true` which causes maps having no `map_layers` being saved as tabular collection. In a multistep workflow where (creating an empty map first and then adding map layers this would result in a `tabular-collection` independend of the map layer's type). 

However, the change repeats some code. It would be just enough to wrap statement: 

```
instance = serializer.save(
    subtype="tabular-collection" if tabular_collection else None,
)
```